### PR TITLE
bump and fixes for nanoplot

### PR DESCRIPTION
**Pull request by bernt-matthias**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/nanoplot/nanoplot.xml [hunk 1]
```diff
@@ -1,38 +1,60 @@
...
         <requirement type="package" version="@TOOL_VERSION@">nanoplot</requirement>
     </requirements>
+    <stdio>
+        <regex match="kaleido problem" source="stderr" level="fatal" description="No static plots are saved due to some kaleido problem" />
+    </stdio>
     <version_command>NanoPlot --version</version_command>
     <command detect_errors="exit_code"><![CDATA[
-#set $myfiles = $mode.reads.files
+## otherwise kalleido fails with `Less than 64MB of free space in temporary directory for shared memory files: 0`
+export TMPDIR=\$_GALAXY_JOB_TMP_DIR;
```
Comment (bgruening) [inline tools/nanoplot/nanoplot.xml:15]: should that not be set on the job-admin level?

**Review by bgruening — APPROVED** [review]
Other than the small comment, this looks great.

File: tools/nanoplot/nanoplot.xml (hunk 1 above)
Comment (bernt-matthias) [reply tools/nanoplot/nanoplot.xml]: Correct.

Comment (bgruening) [reply tools/nanoplot/nanoplot.xml]: I'm not sure why this is needed. Every Admin can set TMPDIR and actually set it by default. Do we really need this, or is this for make the CI work?

File: tools/nanoplot/nanoplot.xml [hunk 2]
```diff
@@ -1,38 +1,61 @@
...
     </requirements>
+    <stdio>
+        <regex match="kaleido problem" source="stderr" level="fatal" description="No static plots are saved due to some kaleido problem" />
+    </stdio>
     <version_command>NanoPlot --version</version_command>
     <command detect_errors="exit_code"><![CDATA[
-#set $myfiles = $mode.reads.files
+## set TMPDIR if not already set by admin
+## otherwise kalleido fails with `Less than 64MB of free space in temporary directory for shared memory files: 0`
+export TMPDIR=\${TMPDIR:-\$_GALAXY_JOB_TMP_DIR};
```
Comment (nsoranzo) [inline tools/nanoplot/nanoplot.xml:16]:
```suggestion
export TMPDIR=\${TMPDIR:-.};
```

File: tools/nanoplot/nanoplot.xml (hunk 1 above)
Comment (bernt-matthias) [reply tools/nanoplot/nanoplot.xml]: Let me try to disable the line and see if the test fails.

> and actually set it by default

Why do you have this assumption?

File: tools/nanoplot/nanoplot.xml (hunk 2 above)
Comment (bernt-matthias) [reply tools/nanoplot/nanoplot.xml:16]: Why not use `$_GALAXY_JOB_TMP_DIR`? I thought this is what it is made for. With `. ` I would be afraid of name collisions.

**Comment by gallardoalba** [comment]
Hi @bernt-matthias, what is missing here? Could I help you?

**Comment by bernt-matthias** [comment]
good to go from my side

**Pull request merged**